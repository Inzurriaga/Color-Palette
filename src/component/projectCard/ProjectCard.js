import React, { Component } from "react";
import { connect } from "react-redux";
import {CopyToClipboard} from 'react-copy-to-clipboard';

export class ProjectCard extends Component{
    render(){
        let { name, palette } = this.props.project
        let palettes = palette.map(color => {
            // change to the index to the id of the palette
            return <div key={color.id}>
                        <h4>{color.name}</h4>
                        <div className="palette-display">
                                    <CopyToClipboard text={color.color1}>
                                        <div style={{"backgroundColor": color.color1}} className="color-display"></div>
                                    </CopyToClipboard>
                                    <CopyToClipboard text={color.color2}>
                                        <div style={{"backgroundColor": color.color2}} className="color-display"></div>
                                    </CopyToClipboard>
                                    <CopyToClipboard text={color.color3}>
                                        <div style={{"backgroundColor": color.color3}} className="color-display"></div>
                                    </CopyToClipboard>
                                    <CopyToClipboard text={color.color4}>
                                        <div style={{"backgroundColor": color.color4}} className="color-display"></div>
                                    </CopyToClipboard>
                                    <CopyToClipboard text={color.color5}>
                                        <div style={{"backgroundColor": color.color5}} className="color-display"></div>
                                    </CopyToClipboard>
                        </div>
                    </div>
        })
        return(
            <article>
                <div className="project-name">
                    <h3>{name}</h3>
                </div>
                <div>
                    {
                        palettes
                    }
                </div>
            </article>
        )
    }
}

export default connect()(ProjectCard)