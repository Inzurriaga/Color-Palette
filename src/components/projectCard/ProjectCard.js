import React, { Component } from "react"
import { connect } from "react-redux"
import {CopyToClipboard} from 'react-copy-to-clipboard'
import { deletePalette } from "../../thunks/deletePalette"
import { deleteProject } from "../../thunks/deleteProject"
import { displayProjectPalette } from "../../action"

export class ProjectCard extends Component{

    deletePalette = (id) => {
        this.props.deletePalette(id)
    }

    deleteProject = () => {
        this.props.deleteProject(this.props.project.id)
    }

    displayPalette = (palette) => {
        let projectPalette = []
        for(var i = 0; i < 5; i++){
            projectPalette.push({
                hex: palette[`color${i+1}`],
                lock: false
            })
        }
        this.props.displayProjectPalette(projectPalette)
    }

    render(){
        let { name, palette } = this.props.project
        let palettes = palette.map(color => {
            return <div className="palette" key={color.id}>
                        <h4 onClick={() => {this.displayPalette(color)}}>{color.name}</h4>
                        <button className="delete-palette" onClick={() => {this.deletePalette(color.id)}}>X</button>
                        <div className="palette-display">
                            <CopyToClipboard text={color.color1}>
                                <div style={{"backgroundColor": color.color1}} className="color-display"></div>
                            </CopyToClipboard>
                            <CopyToClipboard text={color.color2}>
                                <div style={{"backgroundColor": color.color2}} className="color-display"></div>
                            </CopyToClipboard>
                            <CopyToClipboard text={color.color3}>
                                <div style={{"backgroundColor": color.color3}} className="color-display"></div>
                            </CopyToClipboard>
                            <CopyToClipboard text={color.color4}>
                                <div style={{"backgroundColor": color.color4}} className="color-display"></div>
                            </CopyToClipboard>
                            <CopyToClipboard text={color.color5}>
                                <div style={{"backgroundColor": color.color5}} className="color-display"></div>
                            </CopyToClipboard>
                        </div>
                    </div>
        })
        return(
            <article>
                <div className="project-name">
                    <h3>{name}</h3>
                    <button onClick={this.deleteProject}>X</button>
                </div>
                <div>
                    {
                        palettes
                    }
                </div>
            </article>
        )
    }
}

export const mapDispatchToProps = (dispatch) => ({
    deletePalette: id => dispatch(deletePalette(id)),
    deleteProject: id => dispatch(deleteProject(id)),
    displayProjectPalette: palette => dispatch(displayProjectPalette(palette))
})

export default connect(null, mapDispatchToProps)(ProjectCard)