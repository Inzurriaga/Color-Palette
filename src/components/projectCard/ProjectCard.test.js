import React from "react"
import { shallow } from "enzyme"
import { ProjectCard, mapDispatchToProps, mapStateToProps } from "./ProjectCard"

const mockDeletePalette = jest.fn();
const mockDeleteProject = jest.fn()
const mockDisplayProjectPalette = jest.fn()
const mockProject = {
    name: "color",
    id: 5,
    palette: [{
        id: 32,
        color1: "#2a01bd",
        color2: "#4e2b30",
        color3: "#235e2b",
        color4: "#c8604a",
        color5: "#3410f7",
        project_id: 3,
        name: "adsf"
    }]
}

describe("App", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <ProjectCard deletePalette={mockDeletePalette}
                        deleteProject={mockDeleteProject}
                        displayProjectPalette={mockDisplayProjectPalette}
                        project={mockProject}/>
        )
    })
    describe("App", () => {
        it("should match snap shot", () => {
            expect(wrapper).toMatchSnapshot()
        })
    })
    describe("mapDispatchToProps", () => {

    })
    describe("mapStateToProps", () => {

    })
})