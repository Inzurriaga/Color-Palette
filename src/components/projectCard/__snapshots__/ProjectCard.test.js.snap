// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App should match snap shot 1`] = `
<article>
  <div
    className="project-name"
  >
    <h3>
      color
    </h3>
    <button
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      key="32"
    >
      <h4
        onClick={[Function]}
      >
        adsf
      </h4>
      <button
        onClick={[Function]}
      />
      <div
        className="palette-display"
      >
        <CopyToClipboard
          text="#2a01bd"
        >
          <div
            className="color-display"
            style={
              Object {
                "backgroundColor": "#2a01bd",
              }
            }
          />
        </CopyToClipboard>
        <CopyToClipboard
          text="#4e2b30"
        >
          <div
            className="color-display"
            style={
              Object {
                "backgroundColor": "#4e2b30",
              }
            }
          />
        </CopyToClipboard>
        <CopyToClipboard
          text="#235e2b"
        >
          <div
            className="color-display"
            style={
              Object {
                "backgroundColor": "#235e2b",
              }
            }
          />
        </CopyToClipboard>
        <CopyToClipboard
          text="#c8604a"
        >
          <div
            className="color-display"
            style={
              Object {
                "backgroundColor": "#c8604a",
              }
            }
          />
        </CopyToClipboard>
        <CopyToClipboard
          text="#3410f7"
        >
          <div
            className="color-display"
            style={
              Object {
                "backgroundColor": "#3410f7",
              }
            }
          />
        </CopyToClipboard>
      </div>
    </div>
  </div>
</article>
`;
